File: integrations/TS/AI-PREFERENCES/types/interaction.ts
/*
Disciplined AI Software Development Methodology © 2025 by Jay Baleine is licensed under CC BY-SA 4.0 
https://creativecommons.org/licenses/by-sa/4.0/
*/

export interface InteractionRules {
  avoidOverEnthusiasm: boolean;
  forbiddenWords: string[];
  avoidEmDashes: boolean;
  avoidRhetoricalEffects: boolean;
  noUnverifiablePercentageClaims: boolean;
  stayGroundedInAccuracy: boolean;
  useUncertaintyFlag: boolean;
  neverClaimSolutionFound: boolean;
  useAccurateTerminology: boolean;
  documentationFirstPerson: boolean;
  technicalWritingShowBehavior: boolean;
  useSimplePunctuation: boolean;
  noSmallTalk: boolean;
  avoidFriendlyStatements: boolean;
}

export interface TrainingData {
  flagUnfulfillableRequests: boolean;
  neverImplementUnverifiableFeatures: boolean;
  stateCapabilityLimitations: boolean;
}

File: integrations/PY/AI-PREFERENCES/interaction.py
# Disciplined AI Software Development Methodology © 2025 by Jay Baleine is licensed under CC BY-SA 4.0 
# https://creativecommons.org/licenses/by-sa/4.0/
from dataclasses import dataclass
from typing import List

@dataclass
class InteractionRules:
    avoid_enthusiasm: bool = True
    forbidden_words: List[str] = None
    avoid_em_dashes: bool = True
    no_unverifiable_percentages: bool = True
    ground_in_accuracy: bool = True
    uncertainty_flag: str = "⚠️"
    never_claim_solution: bool = True
    accurate_terminology: bool = True
    first_person_documentation: bool = True
    show_observed_behavior: bool = True
    simple_punctuation: bool = True
    no_small_talk: bool = True
    avoid_friendly_statements: bool = True
    
    def __post_init__(self):
        if self.forbidden_words is None:
            self.forbidden_words = [
                "paradigm", "revolutionary", "leader", "innovator", 
                "mathematical precision", "breakthrough", "flagship", 
                "novel", "enhanced", "sophisticated", "advanced", 
                "excellence", "fascinating", "profound"
            ]

@dataclass
class TrainingData:
    flag_unverifiable: str = "🔬"
    never_implement_unverifiable: bool = True
    state_limitations: bool = True